src/test/test.ts

```diff
@@ -253,309 +255,277 @@ suite('Vulcan', function() {
   let bundler: Bundler;
   const inputPath = 'test/html/default.html';
 
-  const preds = dom5.predicates;
-  let doc;
+  let doc: parse5.ASTNode;
 
-  function process(inputPath, done?: Function, bundlerOptions?: any):
-      Promise<parse5.ASTNode> {
+  function bundle(inputPath, bundlerOptions?: any): Promise<parse5.ASTNode> {
     const opts = bundlerOptions || {};
     bundler = new Bundler(opts);
-    let promise = bundler.bundle(inputPath);
-    if (done) {
-      let err, doc;
-      promise = promise.then((document) => {
-        doc = document;
-      }).catch((error) => {
-        err = error;
-      }).then(() => done(err, doc));
-    }
-    return promise;
+    return bundler.bundle(inputPath);
   }
 
   suite('Default Options', function() {
     test('imports removed', function(done) {
       const imports = preds.AND(
           preds.hasTagName('link'), preds.hasAttrValue('rel', 'import'),
           preds.hasAttr('href'), preds.NOT(preds.hasAttrValue('type', 'css')));
-      process(inputPath)
+      bundle(inputPath)
           .then((doc) => {
             assert.equal(dom5.queryAll(doc, imports).length, 0);
             done();
           })
-          .catch((error) => {
-            done(error);
-          });
+          .catch(done);
```

Comment:
Does this successfully report errors? I could have sworn I tried this pattern and it didn't work, but it might have had to do with an incomplete refactor of `bundle(`